pina/geometry/union.py

```diff
@@ -88,11 +95,25 @@ def sample(self, n, mode='random', variables='all'):
                 1000
         """
         sampled_points = []
-        for geometry in self.geometries:
-            points = geometry.sample(
-                int(n/len(self.geometries)), mode, variables)
+        remainder = n % len(self.geometries)
+
+        for i, geometry in enumerate(self.geometries):
+            num_points = n // len(self.geometries)
+            if i < remainder:
+                num_points += 1
+            points = geometry.sample(num_points, mode, variables)
             sampled_points.append(points)
 
         combined_points = torch.cat(sampled_points)
+        return LabelTensor(torch.tensor(combined_points), labels=[f'{i}' for i in self.variables])
```

Comment:
remove line 107 and just call `LabelTensor(torch.cat(sampled_points), labels=[f'{i}' for i in self.variables])`